crates/map/src/finders/dfs.cairo

```diff
@@ -0,0 +1,184 @@
+//! Depth-First Search algorithm implementation for pathfinding.
+
+// Core imports
+use core::dict::{Felt252Dict, Felt252DictTrait};
+
+// Internal imports
+use origami_map::finders::finder::Finder;
+use origami_map::helpers::bitmap::Bitmap;
+use origami_map::helpers::seeder::Seeder;
+use origami_map::types::node::{Node, NodeTrait};
+use origami_map::types::direction::{Direction, DirectionTrait};
+
+/// DepthFirstSearch implementation for pathfinding
+#[generate_trait]
+pub impl DFS of DFSTrait {
```

Comment:
I prefer the long name, to be consistent with BFS https://github.com/dojoengine/origami/blob/07f2b262b2653ba119f8e55eefe68cca18f4dc89/crates/map/src/finders/bfs.cairo#L16